waspc/src/Wasp/Generator/DbGenerator/Jobs.hs

```diff
@@ -27,28 +35,111 @@ migrateDev projectDir = do
   --   we are using `script` to trick Prisma into thinking it is running in TTY (interactively).
 
   -- NOTE(martin): For this to work on Mac, filepath in the list below must be as it is now - not wrapped in any quotes.
-  let npxPrismaCmd = ["npx", "prisma", "migrate", "dev", "--schema", SP.toFilePath schemaFile]
+  let npxPrismaCmd = ["npx", "--no-install", prismaNpxCommand, "migrate", "dev", "--schema", SP.toFilePath schemaFile]
   let scriptArgs =
         if System.Info.os == "darwin"
           then -- NOTE(martin): On MacOS, command that `script` should execute is treated as multiple arguments.
             ["-Fq", "/dev/null"] ++ npxPrismaCmd
           else -- NOTE(martin): On Linux, command that `script` should execute is treated as one argument.
             ["-feqc", unwords npxPrismaCmd, "/dev/null"]
 
-  runNodeCommandAsJob serverDir "script" scriptArgs J.Db
+  let job = runNodeCommandAsJob serverDir "script" scriptArgs J.Db
+
+  retryJobOnErrorWith job (npmInstall projectDir) jobOutputChannelForwarder
 
 -- | Runs `prisma studio` - Prisma's db inspector.
 runStudio :: Path' Abs (Dir ProjectRootDir) -> J.Job
 runStudio projectDir = do
   let serverDir = projectDir </> serverRootDirInProjectRootDir
   let schemaFile = projectDir </> dbSchemaFileInProjectRootDir
 
+  let job =
+        runNodeCommandAsJob
+          serverDir
+          "npx"
+          [ "--no-install",
+            prismaNpxCommand,
+            "studio",
+            "--schema",
+            SP.toFilePath schemaFile
+          ]
+          J.Db
+
+  retryJobOnErrorWith job (npmInstall projectDir) jobOutputChannelForwarder
+
+-- | Runs `prisma generate` to generate the Prisma client.
+runGenerate :: Path' Abs (Dir ProjectRootDir) -> J.Job
+runGenerate projectDir = do
+  let serverDir = projectDir </> serverRootDirInProjectRootDir
+  let schemaFile = projectDir </> dbSchemaFileInProjectRootDir
+
+  let job =
+        runNodeCommandAsJob
+          serverDir
+          "npx"
+          [ "--no-install",
+            prismaNpxCommand,
+            "generate",
+            "--schema",
+            SP.toFilePath schemaFile
+          ]
+          J.Db
+
+  retryJobOnErrorWith job (npmInstall projectDir) errorOnlyJobOutputChannelForwarder
+
+-- | Runs `npm install` to install dependencies.
+-- May be needed before npx commands if dependencies (e.g. Prisma) are not installed.
+npmInstall :: Path' Abs (Dir ProjectRootDir) -> J.Job
```

Comment:
hmm sometimes we use "run" as a prefix and sometimes we don't. Do we have any rule of thumb for it? Not a biggie, just thinking out loud. I am sure it will all become more standardized as we keep working on the code.

Reply:
I think `runStudio` is the only db action we prefix with run atm.